src/main/java/app/sami/languageWeb/contract/ContractService.java

```diff
@@ -0,0 +1,90 @@
+package app.sami.languageWeb.contract;
+
+import app.sami.languageWeb.contract.dtos.StorageUriDto;
+import app.sami.languageWeb.contract.models.Contract;
+import app.sami.languageWeb.contract.models.Status;
+import app.sami.languageWeb.error.exceptions.ContractTranslatedFileAbsent;
+import app.sami.languageWeb.error.exceptions.NotFoundException;
+import app.sami.languageWeb.error.exceptions.UserNotAllowedException;
+import app.sami.languageWeb.request.RequestRepository;
+import app.sami.languageWeb.request.models.Request;
+import app.sami.languageWeb.storage.Storage;
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+@AllArgsConstructor
+public class ContractService {
+
+    private final RequestRepository requestRepository;
+    private final Storage storage;
+    private final ContractRepository contractRepository;
+
+    public StorageUriDto getUploadUri(UUID subject, String fileName){
+        String path = String.format("/contracts/%s/%s__%s", subject, UUID.randomUUID(), fileName);
+        String url = storage.getUploadPresignedUrl(path);
+
+        return StorageUriDto.builder()
+                .url(url)
+                .fileName(path)
+                .build();
+    }
+
+    public Contract createContract(UUID subject, Long requestId){
+        Request request = requestRepository.findById(requestId).orElseThrow(NotFoundException::new);
+        if (request.isUser(subject)){
+            throw new UserNotAllowedException();
+        }
+        requestRepository.save(request.withStatus(app.sami.languageWeb.request.models.Status.ACCEPTED));
```

Comment:
rename `Status` to `RequestStatus` and `ContractStatus`

Reply:
tell don't ask:
instead of `request.withStatus`, prefer `request.accept()`

also, it might be worth to annotate @Transactional on your method so that it only makes a single save call to the db